ui/src/criteria/attribute.tsx

```diff
@@ -323,7 +323,7 @@ function AttributeDetails(props: AttributeDetailsProps) {
     return (
       <>
         {props.data.selected.map(({ value, name }) => (
-          <Stack direction="row" alignItems="baseline" key={Date.now()}>
```

Comment:
Why the change from Date -> String?

Reply:
This was effectively using a random key before, which worked because the list was never visible while being edited. Now the keys actually need to stable of duplicate UI elements can be created.